backtrack/subset.go

```diff
@@ -0,0 +1,38 @@
+// subset.go
+// description: subset of a given set
+// details:
+//In mathematics, set A is a subset of a set B if all elements of A are also elements of B; B is then a superset of A. It is possible for A and B to be equal; if they are unequal, then A is a proper subset of B. The relationship of one set being a subset of another is called inclusion (or sometimes containment). A is a subset of B may also be expressed as B includes (or contains) A or A is included (or contained) in B.
+//  - https://en.wikipedia.org/wiki/Subset
+// author: [sixwaaaay](https://github.com/sixwaaaay)
+// see subset_test.go
+
+package backtrack
+
+func Subset(nums []int) (set [][]int) {
```

Comment:
Do I directly change this to a good one with complex branching and bounds, or is it better to implement a simple (n int) function first, or to limit the currently written function to an array nums of distinct integers
For example, permutation(n int), I can assume that this time the array is only distinct integers

Reply:
Just directly change this to the end goal - no need to go through the intermediate steps since conceptually they will be the same and it would require us maintainers to review more implementations of the same algorithm.